cupyx/jit/_compile.py

```diff
@@ -54,6 +55,53 @@ def new_func(node, *args, **kwargs):
     return new_func
 
 
+def _parse_function_object(func):
+    # Parses function object into ast.FunctionDef object.
+    if not callable(func):
+        raise ValueError('`func` must be a callable object.')
+
+    if jit._getsource_func is not None:
+        full_source = jit._getsource_func(func)
+        start_line, end_line = 0, math.inf
+        source = full_source
+    else:
+        filename = inspect.getsourcefile(func)
+        with open(filename) as f:
+            full_source = f.read()
+        source, start_line = inspect.getsourcelines(func)
+        end_line = start_line + len(source)
+        source = ''.join(source)
+
+    tree = ast.parse(full_source)
+
+    def search(node, instance):
+        if isinstance(node, instance) and start_line <= node.lineno < end_line:
```

Comment:
```suggestion
        if isinstance(node, instance) and (start_line is None or start_line == node.lineno):
```

Reply:
When a single line decorator is attached to the target function, the value of node.line_no will be `start_line + 1`.

Reply:
Ah I see! Thanks for explanation 😄